src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -632,6 +632,31 @@ internal virtual int GetChildPosition(int index)
             return this.Position + offset;
         }
 
+        // Similar to GetChildPosition() but calculating based on the positions of
+        // following siblings rather than previous siblings.
+        internal int GetChildPositionFromEnd(int index)
+        {
+            var green = this.Green;
+            int offset = green.GetSlot(index)?.FullWidth ?? 0;
```

Comment:
>green.GetSlot(index) [](http://example.com/codeflow?start=25&length=20)

It looks like we completely ignore the fact that the item could be cached by now and, therefore, there is no need to iterate.

Reply:
This matches the behavior of `GetChildPosition()` above. It looks like `GetChildPosition()` for a syntax list is typically called when creating the corresponding red node, which is then cached.